backend/src/controllers/AIAssistant.Controller.ts

```diff
@@ -1,17 +1,72 @@
-import { Request, Response } from "express";
+import { Response, Request as ExpressRequest } from "express";
 import { inject } from "inversify";
 import { controller, httpPost } from "inversify-express-utils";
-import { AIASSISTANTSERVICE_BACKEND_PATH, AIAssistantBackendService, MessageRequest } from "../protocol";
+import { AIASSISTANTSERVICE_BACKEND_PATH, AIAssistantBackendService, OAuthService, MessageRequest } from "../protocol";
 import { validateBody } from "../config/validateBody-middleware";
+import { DiscussionType, ProjectType, UserType } from "../database/models";
+import DatabaseService from "../services/database-service";
+
+interface Request extends ExpressRequest {
+  user: UserType;
+  body: any;
+}
 
 @controller(AIASSISTANTSERVICE_BACKEND_PATH)
 export class AIAssistantController {
-  constructor() { }
   
   @inject(AIAssistantBackendService) private aiAssistant: AIAssistantBackendService;
+  @inject(OAuthService) private oAuthService: OAuthService;
+  @inject(DatabaseService) private databaseService: DatabaseService;
+
   @httpPost("/", validateBody(MessageRequest))
   async getAnswer(req: Request, res: Response) {
-    const response = await this.aiAssistant.getAnswer(req.body);
-    return res.json(response);
+    try {
+      const { projectName } = req.body;
+      
+      const user = await this.getUser(req);
+      const project = await this.getProject(projectName);
+      const discussion = await this.getDiscussion(user, project);
+      
+      // Add user message to database
+      const messageText = req.body.messages[req.body.messages.length - 1].content;
+      this.databaseService.messageService.createMessage(discussion, "user", messageText, null, null);
+      
+      const response = await this.aiAssistant.getAnswer(req.body);
+      if (response.content.content === null) {
+        throw new Error("The response from the backend service is null");
+      }
+      
+      // Add backend response message to database
+      this.databaseService.messageService.createMessage(discussion, "assistant", response.content.content, null, null);
+      
+      return res.json(response);
+    } catch (error) {
+      console.log(error);
+      return res.status(500).json({ error: error.message });
+    }
```

Comment:
I am wondering if we should avoid this try-catch block. The middleware handles error responses. So if an error is thrown and not handled in a service the middleware automatically returns a message with code 500, but we can also give other responses. With this block, we can only give the code 500. Also, errors are automatically logged. See this [config](https://github.com/eclipsesource/ide-ai-assistant/blob/e1c17c167c2e46e5501dba9c983c8a946800303c/backend/src/config/server.config.ts#L22C5-L36C8).

As an example, different errors can appear in the [openAI call](https://github.com/eclipsesource/ide-ai-assistant/blob/e1c17c167c2e46e5501dba9c983c8a946800303c/backend/src/services/openAI-service.ts#L48C14-L48C14). I was thinking we would just give the user what ever code we received from the api call. Such as if we get unauthorized response, the user would also get this. Maybe in this example we could also give a 500, I am not entirely sure which is best. I still think we should use the middleware to handle the errors, what do you think?

Reply:
I didn't know that's how it works, but yeah of course then it would make sense! I'm not sure entirely how it works could you do it then?

Reply:
Yeah, sure I will do it today!